src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -100,17 +108,38 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
                 while (userName.Length == 0);
             }
 
-            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName
-            );
+            passwordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_Password, userName);
 
             //
             // now, prompt for the password
             //
-            WriteToConsole(passwordPrompt, true);
-            password = ReadLineAsSecureString();
-            if (password == null)
+            if (confirmPassword)
+            {
+                retypePasswordPrompt = StringUtil.Format(ConsoleHostUserInterfaceSecurityResources.PromptForCredential_ConfirmPassword, userName);
+                bool passwordsMatch;
+                do
+                {
+                    WriteToConsole(passwordPrompt, true);
+                    password = ReadLineAsSecureString();
+                    WriteToConsole(retypePasswordPrompt, true);
+                    retypedPassword = ReadLineAsSecureString();
```

Comment:
If you look "else" block there is a null check that is Ctrl-C. I'd expect the same logic here too.